ci.sh

```diff
@@ -37,7 +37,7 @@ python -c "import sys, struct, ssl; print('python:', sys.version); print('versio
 echo "::endgroup::"
 
 echo "::group::Install dependencies"
-python -m pip install -U pip $(grep "uv==" test-requirements.txt)
+python -m pip install -U pip "$(grep 'uv==' test-requirements.txt)"
```

Comment:
We can use a similar technique to https://github.com/python-trio/trio/pull/3061 and use `python -m pip install pip uv -c test-requirements.txt`.

Reply:
First off, uv implements constraints through `uv pip -c constraints.txt`.
[I was looking into what exactly constraints do](https://stackoverflow.com/questions/34645821/pip-constraints-files), and from what I can tell, the way we are using uv, installing from a locked dependencies file, there is no reason to use constraints. They seem purely limit the version of a dependency that is able to be installed, and if we are specifying and pinning all dependencies with the lock file, I'm pretty sure using constraints shouldn't be necessary.
Please correct me if I am wrong, I could be misunderstanding something.

Reply:
I'm only recommending using constraints when initially installing `uv` rather than grepping the test requirements for consistency reasons.

Reply:
Changed in bc3a94e